include/rocksdb/advanced_options.h

```diff
@@ -1105,6 +1105,19 @@ struct AdvancedColumnFamilyOptions {
   // additional key comparison during memtable lookup.
   bool paranoid_memory_checks = false;
 
+  // When an iterator scans this number of tombstones from the active memtable
+  // for a single iterator operation, we will try to flush the memtable.
+  // Currently forward scan is supported (SeekToFirst(), Seek() and Next()).
+  // This helps to reduce the overhead of scanning through a large number of
+  // tombstones in memtable.
+  // Users should consider enable deletion-triggered-compaction (see
+  // CompactOnDeletionCollectorFactory) together with this option to compact
+  // away tombstones after the memtable is flushed.
+  //
+  // Default: 0 (disabled)
+  // Dynamically changeable through the SetOptions() API.
+  uint32_t memtable_tombstone_scan_limit = 0;
```

Comment:
Not huge fan of this name, but alternatives like `memtable_tombstone_scan_flush_trigger` might be too long

Reply:
How about `tombstone_scan_flush_trigger`? `flush` is probably enough to suggest that this is for memtable.

Reply:
Decided to use `tombstone_scan_flush_trigger` and not consider overwritten keys for now, as the number of overwritten keys per user key can be capped by option `max_sequential_skip_in_iterations`, and the use case we are targeting also doesn't have much overwritten keys.

Reply:
If we generalize this, perhaps memtable_op_scan_flush_trigger.

Another thing to consider for the future is whether we want to have a threshold for the *average* number of invisible entries scanned over entries returned to the user, over the life of the iterator. In theory, several large stretches of invisible entries is just as bad as one giant stretch.

@cbi42

Reply:
I've generalized the option to include overwritten puts (see db_iter.cc) and updated the option to `memtable_op_scan_flush_trigger`.

> a threshold for the average number of invisible entries scanned over entries returned to the user, over the life of the iterator

Yes, I've thought about this and I think it will be useful.